src/main/webapp/portal/main/partials/access-denied.html

```diff
@@ -0,0 +1,13 @@
+<frame-page app-header-title='Access denied' 
```

Comment:
Maybe I'm misunderstanding how a frame-based application is intended to relate to this template.  Is the typical frame-based application intended to override this `access-denied.html` with something tweaked to provide the details @keirserrie mentions as nice-to-have, the name of the application, the link to the related service webpage and/or the app directory entry?

Should there be an optional `access-denied-config.json` where the frame-based application provides that data as, well, structured data, and that way they're not overriding an HTML template to get those details in?  And then this template gracefully uses whatever's provided in that config and falls back on this generic experience if it ain't got no details?